library/Booster/Pattern/Rewrite.hs

```diff
@@ -397,10 +399,12 @@ data RewriteFailed k
       RuleConditionUnclear (RewriteRule k) Predicate
     | -- | A rewrite rule does not preserve definedness
       DefinednessUnclear (RewriteRule k) Pattern [NotPreservesDefinednessReason]
-    | -- | A unification produced a non-match substitution
+    | -- | A matching produced a non-match substitution
       UnificationIsNotMatch (RewriteRule k) Term Substitution
```

Comment:
Probably need to rename the `UnificationIsNotMatch` constructor to just `IsNotMatch`.